### DISCUSSION TITLE
What is the relationship between the arg "periods" in make_future_dataframe() and "n_forecasts" in model

### DISCUSSION THREAD

**Discussion by HamletMonkey** [ourownstory/neural_prophet#943] [Q&A - get help using NeuralProphet]
I created a model `m` train on daily data from Jan-Jun20.

```python
m = NeuralProphet(
    growth = "off",
    yearly_seasonality=False,
    weekly_seasonality=False,
    daily_seasonality=False,
    n_lags=2,
    learning_rate=0.003,
)
```

I am trying to use `m.make_future_dataframe()` to forecast 3 months ahead (Jul-Sep20). However, passing in different value for the `periods` argument does not change the number of future forecast - I'm only getting the date 2020-07-01 regardless. Am I missing out something here :( ?

![image](https://user-images.githubusercontent.com/85542530/200727469-c5dcdaf6-b9e8-4e02-8119-26ace135aa8a.png)

Adding in `n_forecasts=30` to the model gives me the 3-months `future` dataframe (Jul-Sep20), but am still struggling to wrap my head around the [Prediction Collection](https://neuralprophet.com/collect_predictions.html) concept of yhat(s) here, and are they even related?

**Comment by poo5297** [comment]
Did you find out an explaination to this?

**Reply by SeanKski** [reply]
I'm also struggling with this...

**Comment by skant626** [comment]
I have the same question...

**Comment by mmangione** [comment]
You guys also need to specify how many historic predictions you are going to use. 

My code for this looks like:
`future = m.make_future_dataframe(observed_df, periods=n_days_forecasts, n_historic_predictions=len(observed_df))`

FYI: n_historic_predictions >= n_lags.

**Comment by sdegeye** [comment]
> I created a model `m` train on daily data from Jan-Jun20.
> 
> ```python
> m = NeuralProphet(
>     growth = "off",
>     yearly_seasonality=False,
>     weekly_seasonality=False,
>     daily_seasonality=False,
>     n_lags=2,
>     learning_rate=0.003,
> )
> ```
> 
> I am trying to use `m.make_future_dataframe()` to forecast 3 months ahead (Jul-Sep20). However, passing in different value for the `periods` argument does not change the number of future forecast - I'm only getting the date 2020-07-01 regardless. Am I missing out something here :( ?
> 
> ![image](https://user-images.githubusercontent.com/85542530/200727469-c5dcdaf6-b9e8-4e02-8119-26ace135aa8a.png)
> 
> Adding in `n_forecasts=30` to the model gives me the 3-months `future` dataframe (Jul-Sep20), but am still struggling to wrap my head around the [Prediction Collection](https://neuralprophet.com/collect_predictions.html) concept of yhat(s) here, and are they even related?

**Comment by sdegeye** [comment]
Totally agree with you, the documentation give the focus on the possible plots but the way to manage a long term prediction is not clear.
The process is very slow. I expected that even if we specify a n_forecasts argument of for instance 10 and we need to forecast a complete dataframe of 1000 records in the future that the fonction manage itself the repetition n times of the n_forecast to complete the entire dataframe predictions but it is not the case or maybe I'm wrong? If we have to manage ourself a loop to achieve that, that's not really super.
What's your opinion about that?
